HW1/P6/P6.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from functools import partial
+import findspark
+findspark.init()
+import pyspark
+
+
+def pre_process_data( filename, sc ):
+    wlist = sc.textFile( filename )
+    wlist_words = wlist.flatMap( lambda x: x.split() )
+    wlist_words_filter = wlist_words.filter( lambda x: not( x.isdigit() or x.isupper() or ( x[:-1].isupper() and x[-1:] == "." ) ) )
+    wlist_words_index = wlist_words_filter.zipWithIndex().map( lambda (x,y): (y,x) )
+    wlist_words_index_2 = wlist_words_index
+    size = wlist_words_index_2.count()
+    wlist_words_index_2 = wlist_words_index_2.map( lambda ( x, y ): ( x-1, y ) if x != 0 else ( size, y ) )
+    wlist_words_2 = wlist_words_index.join( wlist_words_index_2 )
+    wlist_words_index_2 = wlist_words_index_2.map( lambda ( x, y ): ( x-1, y ) if x != 0 else ( size, y ) )
```

Comment:
Ray did suggest this as a possible way to avoid running out of words in the case that the last two words didn't appear together anywhere else in the corpus (which could cause your sentences to end prematurely).